### PR TITLE
Save RCL8 Structures

### PR DESCRIPTION
**Pull request by Spedwards**
Add Room properties for observer, nuker, and power spawn.

This contains no functionality.

### CODE REVIEW

**Comment by Dasku** [comment]
Need to load the save memory lines too. 

Will see if I can update from work

**Comment by Spedwards** [comment]
@Dasku, you mean the things I just pushed?

**Comment by Dasku** [comment]
Yep, cheers

File: room.js [hunk 1]
```diff
@@ -1004,6 +1040,21 @@ mod.extend = function(){
             this.memory.spawns = _.map(spawns, id);
         } else this.memory.spawns = [];
     };
+    Room.prototype.saveObserver = function() {
+        [this.memory.observer] = this.find(FIND_MY_STRUCTURESS, {
```
Comment (karlthepagan) [inline room.js]: `FIND_MY_STRUCTURESS` is undef ;)

**Comment by Dasku** [comment]
Also can we please add Power Banks to this?

File: room.js (hunk 1 above)
Comment (Spedwards) [inline room.js]: Nice catch.

**Comment by Spedwards** [comment]
Power Bank added @Dasku, and request filled @karlthepagan.

File: room.js [hunk 2]
```diff
@@ -389,6 +425,23 @@ mod.extend = function(){
                 return this._sources;
             }
         },
+        'powerBank': {
+            configurable: true,
+            get: function() {
+                if (_.isUndefined(this.memory.powerBank)) {
+                    [this._powerBank] = this.find(FIND_STRUCTURES, {
```
Comment (karlthepagan) [inline room.js]: the powerbank needs a game time expiration added

Comment (Spedwards) [inline room.js]: I'm not sure what you mean. Power Banks already have a `ticksToDecay` property.

Comment (Dasku) [inline room.js]: I don't even think decay time is important. I can already see the power in the powerBank with this and if we find them from observers we will always have >4950 ticks which is more than enough.

Comment (logxen) [inline room.js]: @dasku 4500 but yeah still plenty of time

@spedwards neat thanks ... I was trying to figure out what the ticksToDecay property's name is. :P

Comment (karlthepagan) [inline room.js]: The object in memory must know when to be garbage collected.

**Comment by karlthepagan** [comment]
@logxen https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/pull/817 is on hold

I encourage you to merge this change into the needs refactor so that both can move forward. This is the smaller change and has less risk.

The right management decision is to accept this PR and merge it into the higher risk feature.

**Comment by logxen** [comment]
Ok this merged fine, sorry for the trouble. I should have tried that first. I was just worried because I already had these implemented with a bunch of code on top of them already. And this clearly states that there is no actual function here. :P

**Comment by karlthepagan** [comment]
I'm still not 100% fine with how powerBanks are expired, but I'm going to move this into dev so we can focus on that problem specifically.

**Comment by logxen** [comment]
Ugh nope apparently this is broken. Nukers and Powerspawns are not being saved in the same format as everything else. The memory stucture is `<structureType> : [ <structureId> ]` but it is only population '<structurType> : <structureId>` so pulling this in and dropping the existing implementations I had broke it. Any chance you can fix the implementation? Otherwise I need to add special handlers for the non-standard memory entries. :P

**Pull request merged**